pkg/rancher-components/src/components/Form/ArrayList/ArrayList.vue

```diff
@@ -1,111 +1,115 @@
-<script>
-import debounce from 'lodash/debounce';
-import { _EDIT, _VIEW } from '@shell/config/query-params';
-import { removeAt } from '@shell/utils/array';
-import { TextAreaAutoGrow } from '@components/Form/TextArea';
-import { clone } from '@shell/utils/object';
-import { LabeledInput } from '@components/Form/LabeledInput';
-
-const DEFAULT_PROTIP = 'Tip: Paste lines into any list field for easy bulk entry';
-
-export default {
+<script lang="ts">
+import Vue, { PropType } from 'vue'
+import debounce from "lodash/debounce";
+import { _EDIT, _VIEW } from "@shell/config/query-params";
+import { removeAt } from "@shell/utils/array";
+import TextAreaAutoGrow from '@components/Form/TextArea/TextAreaAutoGrow.vue';
+import { clone } from "@shell/utils/object";
+import LabeledInput from "@components/Form/LabeledInput/LabeledInput.vue";
+import { ArrayListData, ArrayListRow } from "./types";
+
+const DEFAULT_PROTIP =
+  "Tip: Paste lines into any list field for easy bulk entry";
+
+export default Vue.extend({
   components: { TextAreaAutoGrow, LabeledInput },
 
   props: {
     value: {
-      type:    Array,
-      default: null,
+      type: Array as PropType<Array<string>>,
+      default: () => { return [] },
     },
     mode: {
-      type:    String,
+      type: String,
       default: _EDIT,
     },
     initialEmptyRow: {
-      type:    Boolean,
+      type: Boolean,
       default: false,
     },
 
     title: {
-      type:    String,
-      default: ''
+      type: String,
+      default: "",
     },
     protip: {
-      type:    [String, Boolean],
+      type: [String, Boolean],
       default: DEFAULT_PROTIP,
     },
     showHeader: {
-      type:    Boolean,
+      type: Boolean,
       default: false,
     },
 
     valueLabel: {
-      type:    String,
-      default: 'Value',
+      type: String,
+      default: "Value",
     },
     valuePlaceholder: {
-      type:    String,
-      default: 'e.g. bar'
+      type: String,
+      default: "e.g. bar",
     },
     valueMultiline: {
-      type:    Boolean,
+      type: Boolean,
       default: false,
     },
 
     addLabel: {
       type: String,
       default() {
-        return this.$store.getters['i18n/t']('generic.add');
+        return this.$store.getters["i18n/t"]("generic.add");
```

Comment:
This is the first component that provides internationalization of default values for labels. I agree with the recommended approach. In addition to this, we will also need to bundle any supported locales along with the rest of the library. 

One more thing, the reliance on the store here is potentially problematic as we have no guarantee that the consumer of this library will have the `i18n/t` getter. We will need to figure out the appropriate refactor to remove this dependency before moving forward.

Reply:
Phil sent me a wrapper component that has the dependency on the store. 

@rak-phillip Does this now need conditional logic so that if you are Rancher, you get the wrapped component, but if you are not Rancher, you get the unwrapped component?

```
export { default as ArrayList } from './ArrayListWrapper.vue';
```

Reply:
I'm having a problem with getting the events from ArrayList to get passed to the parent component of the wrapper, or with the value of the parent component getting updated properly.

For example, in the ingress creation form, there's an ArrayList to the right on this Certificates tab:
<img width="796" alt="Screenshot 2022-11-17 at 11 13 11 PM" src="https://user-images.githubusercontent.com/20599230/202634288-13c13e80-32f3-47de-8cf0-e2c206d3836c.png">

When it works properly, it's supposed to update the YAML like this:

```yaml
  tls:
    - hosts:
        - ioeioei
        - iaieoi245
```
But when the component is in the wrapper, it doesn't update the YAML. I can't see why yet.